src/coreComponents/mesh/utilities/ComputationalGeometry.hpp

```diff
@@ -520,24 +480,15 @@ real64 tetVolume( real64 const X[][3] )
  */
 GEOSX_HOST_DEVICE
 inline
-real64 wedgeVolume( real64 const X[][3] )
+real64 wedgeVolume( real64 const (&X)[6][3] )
 {
-  real64 const tet1[4][3] = { LVARRAY_TENSOROPS_INIT_LOCAL_3( X[0] ),
-                              LVARRAY_TENSOROPS_INIT_LOCAL_3( X[1] ),
-                              LVARRAY_TENSOROPS_INIT_LOCAL_3( X[2] ),
-                              LVARRAY_TENSOROPS_INIT_LOCAL_3( X[4] ) };
-
-  real64 const tet2[4][3] = { LVARRAY_TENSOROPS_INIT_LOCAL_3( X[0] ),
-                              LVARRAY_TENSOROPS_INIT_LOCAL_3( X[2] ),
-                              LVARRAY_TENSOROPS_INIT_LOCAL_3( X[4] ),
-                              LVARRAY_TENSOROPS_INIT_LOCAL_3( X[5] ) };
-
-  real64 const tet3[4][3] = { LVARRAY_TENSOROPS_INIT_LOCAL_3( X[0] ),
-                              LVARRAY_TENSOROPS_INIT_LOCAL_3( X[3] ),
-                              LVARRAY_TENSOROPS_INIT_LOCAL_3( X[4] ),
-                              LVARRAY_TENSOROPS_INIT_LOCAL_3( X[5] ) };
-
-  return tetVolume( tet1 ) + tetVolume( tet2 ) + tetVolume( tet3 );
+  constexpr integer numQuadraturePoints = 6;
+  real64 result =  finiteElement::H1_Wedge_Lagrange1_Gauss6::transformedQuadratureWeight( 0, X );
+  for( localIndex q=1; q<numQuadraturePoints; ++q )
+  {
+    result = result + finiteElement::H1_Wedge_Lagrange1_Gauss6::transformedQuadratureWeight( q, X );
+  }
+  return result;
```

Comment:
All these functions are identical now, the only difference being the type of finite element used, and parameters that are derived from it. Would you mind extracting a function, say
```c++
template< typename FE_TYPE >
GEOSX_HOST_DEVICE inline 
real64 elementVolume( real64 const (&X)[FE_TYPE::numNodes][3] )
{
  // quadrature loop
}
```
and then delegating to it?
```suggestion
  return elementVolume< H1_Wedge_Lagrange1_Gauss6 >( X );
```

Reply:
Yes, that's better.